addons/spreadsheet/static/src/currency/plugins/currency.js

```diff
@@ -21,10 +25,8 @@ export class CurrencyPlugin extends OdooUIPlugin {
 
     constructor(config) {
         super(config);
-        this.dataSources = config.custom.dataSources;
-        if (this.dataSources) {
-            this.dataSources.add(DATA_SOURCE_ID, CurrencyDataSource);
-        }
+        /** @type {import("@spreadsheet/data_sources/server_data").ServerData} */
```

Comment:
```js
/** @type {import("@spreadsheet/data_sources/server_data").ServerData | undefined} */
```

But if `serverData` is undefined, all the functions of this plugin will crash. IMHO we should throw if `config.custom.dataSources` is undefined. It should not happen but it will make the code more robust.
In the previous version, there were checks to ensure that `this.dataSources` was not undefined.